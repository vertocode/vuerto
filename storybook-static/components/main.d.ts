import type { App } from './vue/dist/vue.esm-bundler.js';
import { Spinner, BaseButton } from '../components';
declare const _default: {
    install: (app: App) => void;
};
export default _default;
export { BaseButton, Spinner };
