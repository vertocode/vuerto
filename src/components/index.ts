export { default as BaseButton } from './Button/BaseButton.vue'
export { default as BaseTextField } from './Input/BaseTextField.vue'
export { default as Spinner } from './Spinner/Spinner.vue'
