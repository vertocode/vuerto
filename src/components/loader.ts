import path from "path";
import fs from "fs";

export const getSCSSVariables = (configPath: string) => {
    const userConfigPath = path.resolve(__dirname, configPath)
    let userConfig = {}
    if (fs.existsSync(userConfigPath)) {
        console.log('[vuetage] using your config file')
        userConfig = require(userConfigPath)
        console.log(userConfig)
    } else {
        console.log('[vuetage] not found your config file, using default config of vuetage')
    }

    const scssVariables = Object.entries(userConfig)
        .map(([key, value]) => `--${key.replace(/([A-Z])/g, '-$1').toLowerCase()}: ${value};`)
        .join('\n')

    return `:root { ${scssVariables} }`
}

export const vuetagePlugin = async (configPath: string = 'vuetage.js') => {
    if (document) {
        const scssVariables = getSCSSVariables(configPath)

        const style = document.createElement('style')
        style.textContent = scssVariables
        document.head.appendChild(style)
    } else {
        console.error('[vuetage] document is not defined')
    }
}