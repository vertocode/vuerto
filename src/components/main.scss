/* === Spacing variables === */
$size-tiny-1x: 0.2em;
$size-tiny-2x: 0.4em;
$size-tiny-3x: 0.6em;
$size-small-1x: 0.8em;
$size-small-2x: 1em;
$size-small-3x: 1.2em;
$size-medium-1x: 1.5em;
$size-medium-2x: 1.8em;
$size-medium-3x: 2em;
$size-large-1x: 2.3em;
$size-large-2x: 2.5em;
$size-large-3x: 3em;

/* === Base colors variables === */
$base-color: #F6F6F6;
$label-color: #999999;
$error-color: red;
$disabled-color: #E0E0E0;

/* === Dark colors variables === */
$dark-base-color: #1a1a1a;
$dark-label-color: #CCCCCC;
$dark-error-color: #FF6666;
$dark-disabled-color: #666666;