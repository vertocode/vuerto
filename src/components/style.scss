@import '@/components/variables.scss';

body {
  font-family: 'PT Sans', sans-serif;
  font-weight: 400;
  line-height: 1.2;
}