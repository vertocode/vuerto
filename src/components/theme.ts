import { UserStyleConfig, UserStyleConfigVariables, UserConfig } from '@/typing/Config'


export const getSCSSVariables = (userStyleConfig: UserStyleConfigVariables) => {
    if (!userStyleConfig) {
        console.error('[Vuetage] No user style config found.')
        return ''
    }
    const scssVariables = Object.entries(userStyleConfig)
        .map(([key, value]) => `--${key.replaceAll('_', '-').replace(/([A-Z])/g, '-$1').toLowerCase()}: ${value};`)
        .join('\n')

    return `:root { ${scssVariables} }`
}

export const changeTheme = (userStyleConfig: UserStyleConfig) => {
    if (userStyleConfig?.variables) {
        const scssVariables = getSCSSVariables(userStyleConfig.variables)

        const style = document?.createElement('style')
        if (style?.textContent === undefined) {
            console.error('[Vuetage] No style found.')
            return
        }
        style.textContent = scssVariables
        document.head.appendChild(style)
    }
}

export const defineVuetageConfig = async (userConfig: UserConfig) => {
    if (userConfig?.theme) {
        changeTheme(userConfig.theme)
    }
}