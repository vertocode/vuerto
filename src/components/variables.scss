/* === Spacing variables === */
$size-tiny-1x: 0.2em;
$size-tiny-2x: 0.4em;
$size-tiny-3x: 0.6em;
$size-small-1x: 0.8em;
$size-small-2x: 1em;
$size-small-3x: 1.2em;
$size-medium-1x: 1.5em;
$size-medium-2x: 1.8em;
$size-medium-3x: 2em;
$size-large-1x: 2.3em;
$size-large-2x: 2.5em;
$size-large-3x: 3em;

/* === Base colors variables === */
$base-color: #F6F6F6;
$label-color: #999999;
$error-color: red;
$disabled-color: #E0E0E0;

/* === Dark colors variables === */
$dark-base-color: #1a1a1a;
$dark-label-color: #CCCCCC;
$dark-error-color: #FF6666;
$dark-disabled-color: #666666;

/* === Variant colors === */
// Primary
$primary: #1EA7FD;
$border-primary: #1EA7FDl;
$primary-hover: #0056b3;
$border-primary-hover: #0056b3;

// Success
$success: #28a745;
$border-success: #28a745;
$success-hover: #1e7e34;
$border-success-hover: #1e7e34;

// Danger
$danger: #dc3545;
$border-danger: #dc3545;
$danger-hover: #c82333;
$border-danger-hover: #c82333;

// Dark
$dark: black;
$border-dark: black;
$dark-color: white;
$dark-hover: #424242;
$border-dark-hover: #2f2f2f;

// Outline Primary
$outline-primary: #transparent;
$outline-primary-color: #007bff;
$border-outline-primary: #007bff;
$outline-primary-hover: #007bff;
$outline-primary-color-hover: #fff;

// Outline Success
$outline-success: #transparent;
$outline-success-color: #28a745;
$border-outline-success: #28a745;
$outline-success-hover: #28a745;
$outline-success-color-hover: #fff;

// Outline Danger
$outline-danger: #transparent;
$border-outline-danger: #dc3545;
$outline-danger-color: #dc3545;
$outline-danger-hover: #dc3545;
$outline-danger-color-hover: #fff;

// Outline Dark
$outline-dark: #transparent;
$border-outline-dark: black;
$outline-dark-color: black;
$outline-dark-hover: #969696;
$outline-dark-color-hover: #fff;