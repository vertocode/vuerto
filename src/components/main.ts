import type { App } from 'vue'
import 'font-awesome/css/font-awesome.css'
import './style.scss'
import {
	Spinner,
	BaseButton,
	BaseTextField,
	BaseAutocomplete,
	BaseSelect,
	BaseItem,
	BaseGroup,
	BaseMenu,
	changeTheme
} from '@/components'

export default {
	install: (app: App) => {
		app.component('BaseButton', BaseButton)
		app.component('BaseTextField', BaseTextField)
		app.component('BaseSelect', BaseSelect)
		app.component('BaseAutocomplete', BaseAutocomplete)
		app.component('BaseItem', BaseItem)
		app.component('BaseGroup', BaseGroup)
		app.component('BaseMenu', BaseMenu)
		app.component('Spinner', Spinner)
	}
}

export {
	Spinner,
	BaseButton,
	BaseTextField,
	BaseAutocomplete,
	BaseSelect,
	BaseItem,
	BaseGroup,
	BaseMenu,
	changeTheme
}
