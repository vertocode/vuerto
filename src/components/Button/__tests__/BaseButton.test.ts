// @vitest-environment happy-dom

import { expect, it } from 'vitest'
import { BaseButton } from '@/components'
import { mount } from '@vue/test-utils'

it('should match content snapshot using default template', () => {
	const wrapper = mount(BaseButton)
	expect(wrapper.html()).toMatchSnapshot('__snapshots__/BaseButton.snap')
})

it('should use my label when passed it as slot', () => {
	const label = 'My custom label'
	const wrapper = mount(BaseButton, {
		slots: {
			default: label
		}
	})

	const button = wrapper.find('.base-button')
	expect(button.text()).contains(label)
	expect(wrapper.html()).toMatchSnapshot('__snapshots__/BaseButton.snap')
})

it('should show the button with spinner when the props loading is true', () => {
	const wrapper = mount(BaseButton, {
		props: {
			loading: true
		}
	})

	const spinner = wrapper.find('[data-test-id="spinner"]')
	expect(spinner.exists()).toBe(true)
	expect(wrapper.html()).toMatchSnapshot('__snapshots__/BaseButton.snap')
})

it('should show leftIcon and rightIcon when it is passed as parameter', () => {
	const wrapper = mount(BaseButton, {
		props: {
			leftIcon: 'fa fa-check',
			rightIcon: 'fa fa-whatsapp'
		}
	})

	const leftIcon = wrapper.find('[data-test-id="left-icon"]')
	const rightIcon = wrapper.find('[data-test-id="right-icon"]')
	expect(leftIcon.exists()).toBe(true)
	expect(rightIcon.exists()).toBe(true)
	expect(wrapper.html()).toMatchSnapshot('__snapshots__/BaseButton.snap')
})

it('should have attribute disabled when the disabled props is true', () => {
	const wrapper = mount(BaseButton, {
		props: {
			disabled: true
		}
	})

	const button = wrapper.find('.base-button')
	expect(button.attributes('disabled')).toBe('')
	expect(wrapper.html()).toMatchSnapshot('__snapshots__/BaseButton.snap')
})

it('should be a small button when passed the size as small', () => {
	const wrapper = mount(BaseButton, {
		props: {
			size: 'small'
		}
	})

	const button = wrapper.find('.base-button')
	expect(button.element.classList.contains('btn-small')).toBe(true)
	expect(wrapper.html()).toMatchSnapshot('__snapshots__/BaseButton.snap')
})

it('should be a medium button when passed the size as medium', () => {
	const wrapper = mount(BaseButton, {
		props: {
			size: 'medium'
		}
	})

	const button = wrapper.find('.base-button')
	expect(button.element.classList.contains('btn-medium')).toBe(true)
	expect(wrapper.html()).toMatchSnapshot('__snapshots__/BaseButton.snap')
})

it('should be a large button when passed the size as large', () => {
	const wrapper = mount(BaseButton, {
		props: {
			size: 'large'
		}
	})

	const button = wrapper.find('.base-button')
	expect(button.element.classList.contains('btn-large')).toBe(true)
	expect(wrapper.html()).toMatchSnapshot('__snapshots__/BaseButton.snap')
})