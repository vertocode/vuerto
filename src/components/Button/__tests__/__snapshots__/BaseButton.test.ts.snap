// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should be a large button when passed the size as large > __snapshots__/BaseButton.snap 1`] = `"<div data-v-44e60b58=\\"\\" style=\\"width: max-content;\\"><button data-v-44e60b58=\\"\\" class=\\"base-button btn-large btn-primary base-button-transition base-button-hover\\" title=\\"\\"><span data-v-44e60b58=\\"\\" class=\\"base-button-content\\"><!--v-if-->Label<!--v-if--></span></button></div>"`;

exports[`should be a medium button when passed the size as medium > __snapshots__/BaseButton.snap 1`] = `"<div data-v-44e60b58=\\"\\" style=\\"width: max-content;\\"><button data-v-44e60b58=\\"\\" class=\\"base-button btn-medium btn-primary base-button-transition base-button-hover\\" title=\\"\\"><span data-v-44e60b58=\\"\\" class=\\"base-button-content\\"><!--v-if-->Label<!--v-if--></span></button></div>"`;

exports[`should be a small button when passed the size as small > __snapshots__/BaseButton.snap 1`] = `"<div data-v-44e60b58=\\"\\" style=\\"width: max-content;\\"><button data-v-44e60b58=\\"\\" class=\\"base-button btn-small btn-primary base-button-transition base-button-hover\\" title=\\"\\"><span data-v-44e60b58=\\"\\" class=\\"base-button-content\\"><!--v-if-->Label<!--v-if--></span></button></div>"`;

exports[`should have attribute disabled when the disabled props is true > __snapshots__/BaseButton.snap 1`] = `"<div data-v-44e60b58=\\"\\" style=\\"width: max-content;\\"><button data-v-44e60b58=\\"\\" class=\\"base-button btn-medium btn-primary base-button-transition base-button-hover\\" disabled=\\"\\" title=\\"This button is disabled.\\"><span data-v-44e60b58=\\"\\" class=\\"base-button-content\\"><!--v-if-->Label<!--v-if--></span></button></div>"`;

exports[`should match content snapshot using default template > __snapshots__/BaseButton.snap 1`] = `"<div data-v-44e60b58=\\"\\" style=\\"width: max-content;\\"><button data-v-44e60b58=\\"\\" class=\\"base-button btn-medium btn-primary base-button-transition base-button-hover\\" title=\\"\\"><span data-v-44e60b58=\\"\\" class=\\"base-button-content\\"><!--v-if-->Label<!--v-if--></span></button></div>"`;

exports[`should show leftIcon and rightIcon when it is passed as parameter > __snapshots__/BaseButton.snap 1`] = `"<div data-v-44e60b58=\\"\\" style=\\"width: max-content;\\"><button data-v-44e60b58=\\"\\" class=\\"base-button btn-medium btn-primary base-button-transition base-button-hover\\" title=\\"\\"><span data-v-44e60b58=\\"\\" class=\\"base-button-content\\"><i data-v-44e60b58=\\"\\" class=\\"fa fa-check\\" data-test-id=\\"left-icon\\"></i>Label<i data-v-44e60b58=\\"\\" class=\\"fa fa-whatsapp\\" data-test-id=\\"right-icon\\"></i></span></button></div>"`;

exports[`should show the button with spinner when the props loading is true > __snapshots__/BaseButton.snap 1`] = `"<div data-v-44e60b58=\\"\\" style=\\"width: max-content;\\"><button data-v-44e60b58=\\"\\" class=\\"base-button btn-medium btn-primary base-button-transition base-button-hover\\" title=\\"\\"><span data-v-44e60b58=\\"\\"><div data-v-44e60b58=\\"\\" class=\\"spinner spinner-small\\" data-test-id=\\"spinner\\"><div></div><div></div><div></div></div></span></button></div>"`;

exports[`should use my label when passed it as slot > __snapshots__/BaseButton.snap 1`] = `"<div data-v-44e60b58=\\"\\" style=\\"width: max-content;\\"><button data-v-44e60b58=\\"\\" class=\\"base-button btn-medium btn-primary base-button-transition base-button-hover\\" title=\\"\\"><span data-v-44e60b58=\\"\\" class=\\"base-button-content\\"><!--v-if-->My custom label<!--v-if--></span></button></div>"`;
